File: wd-june/ruby/word-frequency.rb
players = "virat virat virat sachin sachin rahul dhoni dhoni"
virat_count = 0
sachin_count = 0
rahul_count = 0
dhoni_count = 0
names = players.split(" ")
virat_count = names.count("virat")
sachin_count = names.count("sachin")
rahul_count = names.count("rahul")
dhoni_count = names.count("dhoni")
puts "Virat - #{"*" * virat_count}"
puts "Sachin - #{"*" * sachin_count}"
puts "Rahul - #{"*" * rahul_count}"
puts "Dhoni - #{"*" * dhoni_count}"

File: wd-june/js/day-4-arrays-objects/word-frequency.js
Array.prototype.count = function(n) {
  var count = 0
  this.forEach(function(num){
    if(num == n){
      count++;
    }
  });
  return count;
};

Array.prototype.uniq = function() {
  var result = [];
  this.forEach(function(num){
    if(result.find(function(n){ return n == num }) == undefined){
      result.push(num);
    }
  });
  return result;
};
var str = "virat virat sachin sachin dhoni virat";
var players = str.split(" ");
var players_hash = {};
players.uniq().forEach(function(player){
  players_hash[player] = players.count(player);
});
for(var key in players_hash){
  console.log(key + " " + "*".repeat(players_hash[key]));
}